src/binheap.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding utf-8 -*-
+
+
+class Binheap(object):
+
+    def __init__(self, iterable=None):
+        '''Initialise a heap from an iterable or an empty heap'''
+        if iterable is None:
+            self.heap = None
+        else:
+            self.heap = sorted(iterable)
+
+    def push(self, val):
+        '''puts new value ito the heap, maintaining the heap property'''
+        if val is None:
+            raise ValueError('push requires a numeric value')
+        elif self.heap is None:
```

Comment:
Your only check for if the heap is empty is if it is `None`.

What would/should happen if you `pop` your heap down to empty, then try to push again?